liota/core/package_manager.py

```diff
@@ -518,14 +515,23 @@ def _package_load(self, file_name, ext_forced=None, check_stack=None):
             log.warning("Package already loaded: %s" % file_name)
             return None
 
-        path_file_ext, file_ext = self._package_chk_exists(
+        path_file_ext, file_ext, checksum_list = self._package_chk_exists(
```

Comment:
You can NOT calculate checksum twice from the file and just compare it. It will ALWAYS be the same.

For packages being loaded using 'load', 'update', 'reload' etc commands over pipe, you must receive the package hash along with file name and pass it to _package_load along-with filename.

For packages being automatically loaded using pkg_list file at startup, pkg_list file should contain hash values in it alongwith the filenames.